api/src/main/java/org/apache/iceberg/IncrementalTableScan.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+
+package org.apache.iceberg;
+
+/**
+ * API for configuring an incremental table scan
+ */
+public interface IncrementalTableScan extends Scan<IncrementalTableScan> {
+
+  /**
+   * Optional. if not set, null value will be used for the start snapshot id.
+   * That would include the oldest ancestor of the {@link IncrementalTableScan#toSnapshotId(long)},
+   * as its parent snapshot id is null which matches the null start snapshot id
+   *
+   * @param fromSnapshotId the start snapshot id (exclusive)
+   * @return an incremental table scan from {@code fromSnapshotId} exclusive
+   */
+  IncrementalTableScan fromSnapshotId(long fromSnapshotId);
```

Comment:
Will it be more generic to add a `inclusive` flag in this `fromSnapshotId` method ?  (So that we can meet the requirement for both including & excluding fromSnapshotId incremental scan )

Reply:
agree. Base on this, is there should add method like useSnapshot to process just one snapshot. I think it's useful for read iceberg table in streaming mode.

Reply:
actually, the `fromSnapshotId` is exclusive behavior (from, to] for the incremental scan, as the `toSnapshotId will become the `fromSnapshotId` in the next scan.

For the inclusive behavior, we were mainly talking about the starting strategy. E.g., if we said a specific start snapshot id, we want to include the files in this snapshot (if append). To support that, we need the incremental scan to support nullable `fromSnapshotId`, as we will just pass in the parent snapshot id (which can be null). That wasn't possible with the `TableScan#appendsBetween(long fromSnapshotId, long toSnapshotId)`.

Reply:
If we want to add exclusive behavior, then we should add alternative methods like `afterSnapshotId`. To me, `fromSnapshotId` should be inclusive of the snapshot that is identified. We can come up with better names for these, like `fromSnapshotInclusive` and `fromSnapshotExclusive` if you like those better.